### PR TITLE
os/board/esp32_devkit/: split spiram to be independent

### PR DESCRIPTION
**Pull request by SRCN-LiQin**
1.split spiram code from esp idf port code for wlan
2.change directory of spiram code, move to os/board/

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [4cff820fcae066e03d4c6f3a1546bb7e8973f862] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [4cff820fcae066e03d4c6f3a1546bb7e8973f862] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_spi.c:162: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:163: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:164: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:165: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:166: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:167: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:168: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:169: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:170: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:171: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:172: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:173: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:174: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:175: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:176: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:177: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:178: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:179: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:180: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:181: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:VxW)
os/arch/xtensa/src/esp32/esp32_spi.c:260: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/esp32_spi.c:261: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/esp32_spi.c:262: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/esp32_spi.c:263: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/arch/xtensa/src/esp32/esp32_spi.c:410: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/arch/xtensa/src/esp32/esp32_spi.c:1124: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/arch/xtensa/src/esp32/esp32_spi.c:1132: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/board/esp32_devkit/include/spiram_psram.h:54: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/board/esp32_devkit/src/spiram.c:79: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/board/esp32_devkit/src/spiram_psram.c:217: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/board/esp32_devkit/src/spiram_psram.c:217: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
os/board/esp32_devkit/src/spiram_psram.c:234: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
os/board/esp32_devkit/src/spiram_psram.c:269: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/board/esp32_devkit/src/spiram_psram.c:270: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/board/esp32_devkit/src/spiram_psram.c:271: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
os/board/esp32_devkit/src/spiram_psram.c:335: ERROR: [BRC_M_FTN] open brace '{' following function declarations go on the next line
os/board/esp32_devkit/src/spiram_psram.c:347: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:380: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:411: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:454: ERROR: [SPC_M_OPR] spaces required around that '!=' (ctx:VxV)
os/board/esp32_devkit/src/spiram_psram.c:497: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:583: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:637: ERROR: [SPC_M_OPR] need consistent spacing around '|' (ctx:VxW)
os/board/esp32_devkit/src/spiram_psram.c:675: ERROR: [IDT_M_SCH] switch and case should be at the same indent
os/board/esp32_devkit/src/spiram_psram.c:682: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/board/esp32_devkit/src/spiram_psram.c:690: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/board/esp32_devkit/src/spiram_psram.c:699: ERROR: [IDT_M_TAB] code indent should use tabs where possible
os/board/esp32_devkit/src/spiram_psram.c:709: ERROR: [IDT_M_SCH] switch and case should be at the same indent

**Comment by seinfra** [comment]
Target : [25ffae86471bc3cdbff69016c29b1b18437b56a9] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [25ffae86471bc3cdbff69016c29b1b18437b56a9] - Code Rule Check Result:
os/arch/xtensa/src/esp32/esp32_spi.c:162: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:163: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:164: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:165: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:166: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:167: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:168: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:169: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:170: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:171: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:172: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:173: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:174: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:175: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:176: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:177: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:178: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:179: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:180: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)
os/arch/xtensa/src/esp32/esp32_spi.c:181: ERROR: [SPC_M_OPR] spaces prohibited around that ':' (ctx:WxW)

**Comment by seinfra** [comment]
Target : [e174c4f43054e80734fba9856aaed1c422050744] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [e174c4f43054e80734fba9856aaed1c422050744] - Code Rule Check OK.

File: os/arch/xtensa/src/esp32/esp32_spi.c [hunk 1]
```diff
@@ -159,26 +159,26 @@ typedef struct {
...
-		uint32_t reserved12: 5;	/*reserved */
-		uint32_t last_command: 3;	/*In the slave mode it is the value of command. */
-		uint32_t last_state: 3;	/*In the slave mode it is the state of spi state machine. */
-		uint32_t trans_cnt: 4;	/*The operations counter in both the master mode and the slave mode. 4: read-status */
-		uint32_t cmd_define: 1;	/*1: slave mode commands are defined in SPI_SLAVE3.  0: slave mode commands are fixed as: 1: write-status 2: write-buffer and 3: read-buffer. */
-		uint32_t wr_rd_sta_en: 1;	/*write and read status enable  in the slave mode */
-		uint32_t wr_rd_buf_en: 1;	/*write and read buffer enable in the slave mode */
-		uint32_t slave_mode: 1;	/*1: slave mode 0: master mode. */
-		uint32_t sync_reset: 1;	/*Software reset enable, reset the spi clock line cs line and data lines. */
+		uint32_t rd_buf_done:1;	/*The interrupt raw bit for the completion of read-buffer operation in the slave mode. */
```
Comment (ChinaTizenRT) [inline os/arch/xtensa/src/esp32/esp32_spi.c]: accroding to code submiting experience,  split coding rule as separate commit.

**Comment by seinfra** [comment]
Target : [254626565f71d231421093b5f89d3744c7a77546] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [254626565f71d231421093b5f89d3744c7a77546] - Code Rule Check OK.

File: os/arch/xtensa/src/esp32/esp32_spi.c (hunk 1 above)
Comment (SRCN-LiQin) [reply os/arch/xtensa/src/esp32/esp32_spi.c]: fixed, thanks for suggestion

File: os/board/esp32_devkit/src/spiram.c [hunk 2]
```diff
@@ -103,17 +102,17 @@ bool esp_spiram_test()
 			}
 		}
 	}
-	ESP_EARLY_LOGI(TAG,"read spiram over\n");
+	lldbg("read spiram over\n");
```
Comment (sunghan-chang) [inline os/board/esp32_devkit/src/spiram.c]: Let's use llvdbg. Because these logs are not errors.

File: os/board/esp32_devkit/src/spiram_psram.c [hunk 3]
```diff
@@ -36,20 +36,62 @@
...
-#include "soc/efuse_reg.h"
-#include "chip/esp32_gpio.h"
-#include "spi_common.h"
+#include "chip/esp32_efuse_reg.h"
+#include "rom/esp32_gpio.h"
+//#include "spi_common.h"
 #include "periph_ctrl.h"
+#include "assert.h"
 
 #if CONFIG_SPIRAM_SUPPORT
```
Comment (sunghan-chang) [inline os/board/esp32_devkit/src/spiram_psram.c]: This should be `#ifdef`.

File: os/board/esp32_devkit/src/spiram_psram.c [hunk 4]
```diff
@@ -531,7 +572,7 @@ esp_err_t IRAM_ATTR psram_enable(psram_cache_mode_t mode, psram_vaddr_mode_t vad
 	}
 
 	WRITE_PERI_REG(GPIO_ENABLE_W1TC_REG, BIT(PSRAM_CLK_IO) | BIT(PSRAM_CS_IO));   //DISABLE OUPUT FOR IO16/17
-	assert(mode < PSRAM_CACHE_MAX && "we don't support any other mode for now.");
+	ASSERT(mode < PSRAM_CACHE_MAX && "we don't support any other mode for now.");
```
Comment (sunghan-chang) [inline os/board/esp32_devkit/src/spiram_psram.c]: I think " " sentence is unnecessary. I have no idea for that. Could you let me know the purpose of that?

File: os/board/esp32_devkit/src/spiram_psram.c (hunk 3 above)
Comment (SRCN-LiQin) [reply os/board/esp32_devkit/src/spiram_psram.c]: fixed, thanks

File: os/board/esp32_devkit/src/spiram.c (hunk 2 above)
Comment (SRCN-LiQin) [reply os/board/esp32_devkit/src/spiram.c]: updated, use llvdbg here, and use dbg in test function

File: os/board/esp32_devkit/src/spiram_psram.c (hunk 4 above)
Comment (SRCN-LiQin) [reply os/board/esp32_devkit/src/spiram_psram.c]: removed, that's for old version assert, useless for ASSERT here. thanks

**Comment by seinfra** [comment]
Target : [550e0f4be563f607ce970080f2a5e64840e25e3a] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [550e0f4be563f607ce970080f2a5e64840e25e3a] - Code Rule Check OK.

File: os/board/esp32_devkit/src/Makefile [hunk 5]
```diff
@@ -89,6 +93,7 @@ else
   CFLAGS += -I$(ARCH_SRCDIR)/chip/rom
   CFLAGS += -I$(BOARD_DIR)/include
   CFLAGS += -I$(TOPDIR)/board/common
+  CFLAGS += -I$(TOPDIR)/../external/esp_idf_port/include
```
Comment (sunghan-chang) [inline os/board/esp32_devkit/src/Makefile]: Why do you need include of external? In architecture point of view, lower layer should not call higher layer. In TizenRT, external positions at top of os.

File: os/board/esp32_devkit/src/spiram.c [hunk 6]
```diff
@@ -164,7 +161,7 @@ size_t esp_spiram_get_size()
  Before flushing the cache, if psram is enabled as a memory-mapped thing, we need to write back the data in the cache to the psram first,
  otherwise it will get lost. For now, we just read 64/128K of random PSRAM memory to do this.
 */
-void IRAM_ATTR esp_spiram_writeback_cache()
+void IRAM_ATTR esp_spiram_writeback_cache(void)
```
Comment (sunghan-chang) [inline os/board/esp32_devkit/src/spiram.c]: Actually, this kind of changes is fixing violations of coding rules. Could you move to 2nd commit?

**Comment by seinfra** [comment]
Target : [231eb9900bd0aec7cf7912be359e3a64ec31353e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [231eb9900bd0aec7cf7912be359e3a64ec31353e] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [6f2785f570ac9306abf0d3644ed97ea48f8feb35] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [6f2785f570ac9306abf0d3644ed97ea48f8feb35] - Code Rule Check OK.

File: os/board/esp32_devkit/src/spiram.c (hunk 6 above)
Comment (SRCN-LiQin) [reply os/board/esp32_devkit/src/spiram.c]: updated, similar modifications moved to 2nd commit

File: os/board/esp32_devkit/src/Makefile (hunk 5 above)
Comment (SRCN-LiQin) [reply os/board/esp32_devkit/src/Makefile]: fixed, thanks for suggestion
